File: CSharp/EntityFramework/Models/Employee.cs
namespace Models;

public class Employee
{
    public Guid Id { get; set; }

    public Guid PersonId { get; set; }
    public Person? Person { get; set; }

    public int DepartmentId { get; set; }
    public Department? Department { get; set; }

    public Position Position { get; set; }

    public override string ToString()
    {
        return $"Employee: {{ Id: {Id}, Person: {Person}, Department: {Department}, Position: {Position} }}";
    }
}

public enum Position { Boss, Manager, Staffer }

File: FSharp/EntityFramework/Models/Employee.fs
namespace Models

open System

type Position =
  | Boss = 0
  | Manager = 1
  | Staffer = 2

type Employee () =
  // [<Key>]
  member val Id: Guid = Guid.Empty with get, set

  member val PersonId: Guid = Guid.Empty with get, set
  member val Person: Person = Person () with get, set

  member val DepartmentId: int = Int32 () with get, set
  member val Department: Department = Department () with get, set

  member val Position: Position = Position () with get, set

  new (person, department, position) as this =
    Employee ()
    then
      this.Person <- person
      this.Department <- department
      this.Position <- position

  with override this.ToString () =
        $"Employee {{ Id: {this.Id}, Person: {this.Person}, Department: {this.Department}, Pisition: {this.Position} }}"
